jib-core/src/main/java/com/google/cloud/tools/jib/api/Containerizer.java

```diff
@@ -312,11 +318,27 @@ Path getBaseImageLayersCacheDirectory() {
   }
 
   Path getApplicationLayersCacheDirectory() throws CacheDirectoryCreationException {
+    return getApplicationLayersCacheDirectory(
+        () -> Paths.get(System.getProperty("java.io.tmpdir")));
+  }
+
+  @VisibleForTesting
+  Path getApplicationLayersCacheDirectory(Supplier<Path> tmpDirSupplier)
+      throws CacheDirectoryCreationException {
     if (applicationLayersCacheDirectory == null) {
-      // Uses a temporary directory if application layers cache directory is not set.
+      // Create a directory in temp if application layers cache directory is not set.
       try {
-        applicationLayersCacheDirectory = Files.createTempDirectory(null);
-        applicationLayersCacheDirectory.toFile().deleteOnExit();
+        Path systemTmp = tmpDirSupplier.get();
+        Path jibTempCache = systemTmp.resolve(DEFAULT_APPLICATION_CACHE_DIRECTORY_NAME);
+        if (Files.exists(jibTempCache)) {
+          if (!Files.isDirectory(jibTempCache)) {
+            throw new CacheDirectoryCreationException(
+                jibTempCache.toString() + " already exists and is not a directory");
+          }
+        } else {
+          Files.createDirectory(jibTempCache);
```

Comment:
Actually think I can clean up all this logic and just use Files.createDirectories ? Or just lets the cache writer handle this?